dsc/pwsh.profile.resource.ps1

```diff
@@ -0,0 +1,149 @@
+
+    $resource = [PwshResource]::new()
+    $resource.profileType = $profileType
+    $resource.content = $fileExists ? (Get-Content -Path $profilePath -Raw) : $null
+    $resource._exist = $fileExists
+
+    return $resource
+}
+
+function GetProfilePath {
+    param (
+        [ProfileType] $profileType
+    )
+
+    $path = switch ($profileType) {
+        'AllUsersCurrentHost' { $PROFILE.AllUsersCurrentHost}
+        'AllUsersAllHosts' { $PROFILE.AllUsersAllHosts}
+        'CurrentUserAllHosts' { $PROFILE.CurrentUserAllHosts}
+        'CurrentUserCurrentHost' { $PROFILE.CurrentUserCurrentHost}
+    }
+
+    return $path
+}
+
+function ExportOperation {
+    $allUserCurrentHost = PopulatePwshResource -profileType 'AllUsersCurrentHost'
+    $allUsersAllHost = PopulatePwshResource -profileType 'AllUsersAllHosts'
+    $currentUserAllHost = PopulatePwshResource -profileType 'CurrentUserAllHosts'
+    $currentUserCurrentHost = PopulatePwshResource -profileType 'CurrentUserCurrentHost'
+
+    # Cannot use the ToJson() method here as we are adding a note property
+    $allUserCurrentHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'AllUsersCurrentHost' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+    $allUsersAllHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'AllUsersAllHosts' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+    $currentUserAllHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'CurrentUserAllHosts' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+    $currentUserCurrentHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'CurrentUserCurrentHost' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+}
+
+function GetOperation {
+    param (
+        [PwshResource] $InputResource
+    )
+
+    $profilePath = GetProfilePath -profileType $InputResource.profileType.ToString()
+
+    $exists = Test-Path $profilePath
+
+    if ($InputResource._exist -and $exists) {
+        $content = Get-Content -Path $profilePath
+        $InputResource.Content = $content
+    }
+    elseif ($InputResource._exist -and -not $exists) {
+        $InputResource.Content = $null
+        $InputResource._exist = $false
+    }
+    elseif (-not $InputResource._exist -and $exists) {
+        $InputResource.Content = Get-Content -Path $profilePath
+        $InputResource._exist = $true
+    }
+    else {
+        $InputResource.Content = $null
+        $InputResource._exist = $false
+    }
+
+    $InputResource.ToJson()
+}
+
+function SetOperation {
+    param (
+        [PwshResource] $InputResource
+    )
+
+    $profilePath = GetProfilePath -profileType $InputResource.profileType.ToString()
+    $profileExists = Test-Path $profilePath
+
+    if ($InputResource._exist) {
+        if ($InputResource.content) {
+            Set-Content -Path $profilePath -Value $InputResource.content
+        }
+        else {
+            Remove-Item -Path $profilePath -Force
+        }
```

Comment:
This is a little surprising - if I understand correctly, the following snippet will _remove_ the current user current host profile:

```yaml
- type: Microsoft.PowerShell/Profile
  name: Personal PowerShell profile
  properties:
    _exist: true
```

I would expect it to be left alone if it already exists or create an empty file if it doesn't exist.

Reply:
Fixed.